omnigibson/systems/micro_particle_system.py

```diff
@@ -329,31 +309,10 @@ def particle_prototype_ids(self, prototype_ids):
         self.set_attribute(attr="protoIndices", val=prototype_ids)
 
     @property
-    def particle_visibilities(self):
-        """
-        Returns:
-            np.array: (N,) numpy array, where each entry is the specific particle's visibility
-                (1 if visible, 0 otherwise)
-        """
-        # We leverage the ids + invisibleIds prim fields to infer visibility
-        # id = 1 means visible, id = 0 means invisible
-        ids = self.get_attribute("ids")
-        return np.ones(self.n_particles) if ids is None else np.array(ids)
-
-    @particle_visibilities.setter
-    def particle_visibilities(self, visibilities):
-        """
-        Set the particle visibilities for this instancer
-
-        Args:
-            np.array: (N,) numpy array, where each entry is the specific particle's desired visibility
-                (1 if visible, 0 otherwise)
-        """
-        assert visibilities.shape[0] == self._n_particles, \
-            f"Got mismatch in particle setting size: {visibilities.shape[0]}, vs. number of particles {self._n_particles}!"
-        # We leverage the ids + invisibleIds prim fields to infer visibility
-        # id = 1 means visible, id = 0 means invisible
-        self.set_attribute(attr="ids", val=visibilities)
+    def state_size(self):
+        # idn, particle_group, n_particles (3),
+        # and the corresponding states for each particle (N * 14)
+        return 3 + len(self._n_particles) * (3 + 3 + 4 + 3 + 1)
```

Comment:
maybe we just put 14 here and break it down in the comment (3 pos + 3 vel + 4 quat + 3 scale + 1 id)?

Reply:
done